libs/offline-tasks/src/main/java/org/opensearch/offline_tasks/task/Task.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.offline_tasks.task;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+
+/**
+ * A Background Task to be run on Offline Node.
+ */
+@ExperimentalApi
+public class Task {
```

Comment:
Do we need to track the task status as an attribute here?

Reply:
yes, this can be added. Thanks for catching this